cmd/collectors/zapi/plugins/shelf/shelf.go

```diff
@@ -139,51 +144,100 @@ func (my *Shelf) Init() error {
 	}
 
 	my.Logger.Debug().Msgf("initialized with data [%d] objects", len(my.data))
+
+	// setup batchSize for request
+	if my.client.IsClustered() {
+		if b := my.Params.GetChildContentS("batch_size"); b != "" {
+			if _, err := strconv.Atoi(b); err == nil {
+				my.batchSize = b
+			}
+		} else {
+			my.batchSize = BatchSize
```

Comment:
probably better to initialize with `my.batchSize = BatchSize` at line 149. Without that, the logic below results in `my.batchSize` being 0 when the param ` batch_size` has a parsing error. Or maybe you intended that `my.batchSize=0` in that case?

Reply:
initialized with `BatchSize`.